File: donghwa/0x04/1158.cpp
#include <bits/stdc++.h>
using namespace std;

const int MX = 5000;
int res[MX], prv[MX], nxt[MX], dat[MX];
int cursor = 0;

void move(int cnt){
    while (cnt != 0){
        cursor = nxt[cursor];
        cnt--;
    }
}
void dlt(){
    nxt[prv[cursor]] = nxt[cursor];
    prv[nxt[cursor]] = prv[cursor];
    cursor = nxt[cursor];
}


int main(void){
    fill(res, res+MX, -1);
    int N;
    cin >> N;

    for (int i=0; i<N; i++){
        dat[i] = i;
        if (i == 0){
            prv[i] = N-1;
            nxt[i] = i+1;
        } else if (i == N-1){
            prv[i] = N-2;
            nxt[i] = 0;
        } else {
            prv[i] = i-1;
            nxt[i] = i+1;
        }
    }
    int cnt;
    cin >> cnt;
    int idx = 0;
    int M = N;
    while (N != 0){
        move(cnt-1);
        res[idx] = cursor ;
        dlt();
        idx ++;
        N --;
    }
    idx = 0;
    cout << "<";
    while (res[idx] != -1){
        cout << res[idx]+1;
        if (idx != M-1) cout << ", ";
        idx++;
    }
    cout << ">";
}

File: donghwa/0x04/1158.py

def move(cnt):
    global cursor, nxt
    while (cnt != 0) :
        cursor = nxt[cursor]
        cnt -= 1        
    
def dlt() :
    global nxt, prv, cursor
    nxt[prv[cursor]] = nxt[cursor]
    prv[nxt[cursor]] = prv[cursor]
    cursor = nxt[cursor]

if __name__ == "__main__":
    
    N, cnt = list(map(int, input().split()))
    dat = list(range(N))
    prv = dat.copy()
    prv.insert(0,prv.pop(-1))
    nxt = dat.copy()
    nxt.insert(N-1,nxt.pop(0))
    cursor = 0
    res = []
    while (N != 0) :
        while (cnt != 0) :
            cursor = nxt[cursor]
            cnt -= 1     
        
        res.append(cursor+1)
        
        nxt[prv[cursor]] = nxt[cursor]
        prv[nxt[cursor]] = prv[cursor]
        cursor = nxt[cursor]
        N -= 1
    
    prt = "<"
    for idx, item in enumerate(res) :
        if idx != len(res)-1 :
            prt += str(item) + ", "
    prt += str(res[-1]) + ">" 
    print(prt)

